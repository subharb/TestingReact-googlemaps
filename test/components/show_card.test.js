import ShowCard from '../../src/components/show_card';
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


test('ShowCard carga el texto enviado', () => {
    const testInfo = { name: "Life is a Test", image: "http://xiostorage.com/wp-content/uploads/2015/10/test.png",
                    startDate: 'Viernes, 1 de enero de 1979', location: "Teatro de Test" };
    const wrapper = mount(
        <ShowCard name={ testInfo.name } image={ testInfo.image } startDate={ testInfo.startDate } location={ testInfo.location } />
    );
    //Testeamos el contenido del Component
    const h4 = wrapper.find('h4');
    expect(h4.text()).toBe(testInfo.name);

    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(testInfo.image);

    const p = wrapper.find('.card-text');
    expect(p.text()).toBe(testInfo.startDate +" " +testInfo.location);
});

describe('Todo component renders the todo correctly', () => {
    it('renders correctly', () => {
        const testInfo = { name: "Life is a Test", image: "http://xiostorage.com/wp-content/uploads/2015/10/test.png",
        startDate: 'Viernes, 1 de enero de 1979', location: "Teatro de Test" };
        const rendered = renderer.create(
            <ShowCard name={ testInfo.name } image={ testInfo.image } startDate={ testInfo.startDate } location={ testInfo.location } />
            );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
  });