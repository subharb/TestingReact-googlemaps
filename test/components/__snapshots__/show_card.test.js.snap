// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly renders correctly 1`] = `
<div
  className="card show"
  style={
    Object {
      "width": "20rem",
    }
  }
>
  <img
    alt="Life is a Test"
    className="card-img-top"
    height="160"
    src="http://xiostorage.com/wp-content/uploads/2015/10/test.png"
    width="320"
  />
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      Life is a Test
    </h4>
    <p
      className="card-text"
    >
      Viernes, 1 de enero de 1979
       
      <br />
      Teatro de Test
    </p>
    <a
      className="btn btn-primary"
      href="#"
    >
      Comprar
    </a>
  </div>
</div>
`;
